platform/mbed_alloc_wrappers.cpp

```diff
@@ -91,6 +91,9 @@ extern "C" {
 
 extern "C" void * __wrap__malloc_r(struct _reent * r, size_t size) {
     void *ptr = NULL;
+#ifdef MBED_MEM_TRACING_ENABLED
+    mbed_mem_trace_lock();
```

Comment:
The memory trace mutex should be held here, otherwise another thread my have acquired this lock but not the mutex, causing logging on other threads to be dropped. Could `mbed_mem_trace_lock()` include `mem_trace_mutex->lock()` to simply this?

Reply:
I mistakenly assumed that whole `__wrap__malloc_r` is protected by `__malloc_lock` but actually `__real__malloc_r` is protected

Reply:
Sure we could encapsulate `mem_trace_mutex` inside `mbed_mem_trace.c` module but since it's C file we have to use raw OS mutex. Other solution it would be changing this module to cpp and use `SingletonPtr<PlatformMutex>`

Reply:
I'm not sure which solution will be acceptable?